### PR TITLE
Update Data Connectors to use new secrets/parameters format

### PR DESCRIPTION
**Pull request by phillipleblanc**
## 🗣 Description

Updates the rest of the documentation to include the changes made from https://github.com/spiceai/spiceai/issues/1701

## 🔨 Related Issues

https://github.com/spiceai/spiceai/issues/1701

### CODE REVIEW

File: spiceaidocs/docs/components/data-accelerators/postgres/index.md [hunk 1]
```diff
@@ -24,8 +24,7 @@ The connection to PostgreSQL can be configured by providing the following `param
 - `pg_port`: The port of the PostgreSQL server.
 - `pg_db`: The name of the database to connect to.
 - `pg_user`: The username to connect with.
-- `pg_pass_key`: The secret key containing the password to connect with.
-- `pg_pass`: The raw password to connect with, ignored if `pg_pass_key` is provided.
+- `pg_pass`: The password to connect with. Use the [secret replacement syntax](../../secret-stores/index.md) to load the password from a secret store, e.g. `${secrets:my_pg_pass}`.
 - `pg_sslmode`: Optional parameter, specifies the SSL/TLS behavior for the connection, supported values:
```
Comment (digadeesh) [inline spiceaidocs/docs/components/data-accelerators/postgres/index.md]:
```suggestion
- `pg_sslmode`: Optional. Specifies the SSL/TLS behavior for the connection, supported values:
```

File: spiceaidocs/docs/components/data-connectors/clickhouse.md [hunk 2]
```diff
@@ -18,20 +18,18 @@ datasets:
...
 - `clickhouse_db`: The name of the database to connect to.
 - `clickhouse_user`: The username to connect with.
-- `clickhouse_pass_key`: The secret key containing the password to connect with.
-- `clickhouse_pass`: The raw password to connect with, ignored if `clickhouse_pass_key` is provided.
+- `clickhouse_pass`: The password to connect with. Use the [secret replacement syntax](../secret-stores/index.md) to load the password from a secret store, e.g. `${secrets:my_clickhouse_pass}`.
 - `clickhouse_secure`: Optional parameter, specifies the SSL/TLS behavior for the connection, supported values:
   - `true`: (default) This mode requires an SSL connection. If a secure connection cannot be established, server will not connect.
   - `false`: This mode will not attempt to use an SSL connection, even if the server supports it.
-- `clickhouse_connection_timeout`: Optional parameter, specifies the connection timeout in milliseconds.
+- `connection_timeout`: Optional parameter, specifies the connection timeout in milliseconds.
```
Comment (digadeesh) [inline spiceaidocs/docs/components/data-connectors/clickhouse.md]:
```suggestion
- `connection_timeout`: Optional. Specifies the connection timeout in milliseconds.
```

File: spiceaidocs/docs/components/data-connectors/clickhouse.md [hunk 3]
```diff
@@ -18,20 +18,18 @@ datasets:
...
+- `clickhouse_connection_string`: The connection string to use to connect to the Clickhouse server. This can be used instead of providing individual connection parameters. Use the [secret replacement syntax](../secret-stores/index.md) to load the password from a secret store, e.g. `${secrets:my_clickhouse_conn_string}`.
 - `clickhouse_host`: The hostname of the Clickhouse server.
 - `clickhouse_tcp_port`: The port of the Clickhouse server.
 - `clickhouse_db`: The name of the database to connect to.
 - `clickhouse_user`: The username to connect with.
-- `clickhouse_pass_key`: The secret key containing the password to connect with.
-- `clickhouse_pass`: The raw password to connect with, ignored if `clickhouse_pass_key` is provided.
+- `clickhouse_pass`: The password to connect with. Use the [secret replacement syntax](../secret-stores/index.md) to load the password from a secret store, e.g. `${secrets:my_clickhouse_pass}`.
 - `clickhouse_secure`: Optional parameter, specifies the SSL/TLS behavior for the connection, supported values:
```
Comment (digadeesh) [inline spiceaidocs/docs/components/data-connectors/clickhouse.md]:
```suggestion
- `clickhouse_secure`: Optional. Specifies the SSL/TLS behavior for the connection, supported values:
```

File: spiceaidocs/docs/components/data-connectors/mysql.md [hunk 4]
```diff
@@ -12,21 +12,24 @@ To connect to any MySQL database as connector for federated SQL query, specify `
...
 - `mysql_host`: The hostname of the MySQL server.
 - `mysql_tcp_port`: The port of the MySQL server.
 - `mysql_db`: The name of the database to connect to.
-- `mysql_user_key`: The name of the secret key containing the MySQL username.
-- `mysql_user`: The MySQL username.  Ignored if `mysql_user_key` is provided.
-- `mysql_pass_key`: The secret key containing the password to connect with.
-- `mysql_pass`: The raw password to connect with, ignored if `mysql_pass_key` is provided.
+- `mysql_user`: The MySQL username.
+- `mysql_pass`: The password to connect with.
 - `mysql_sslmode`: Optional parameter, specifies the SSL/TLS behavior for the connection, supported values:
```
Comment (digadeesh) [inline spiceaidocs/docs/components/data-connectors/mysql.md]:
```suggestion
- `mysql_sslmode`: Optional. Specifies the SSL/TLS behavior for the connection, supported values:
```

File: spiceaidocs/docs/components/data-connectors/postgres/index.md [hunk 5]
```diff
@@ -22,8 +22,7 @@ The connection to PostgreSQL can be configured by providing the following `param
 - `pg_port`: The port of the PostgreSQL server.
 - `pg_db`: The name of the database to connect to.
 - `pg_user`: The username to connect with.
-- `pg_pass_key`: The secret key containing the password to connect with.
-- `pg_pass`: The raw password to connect with, ignored if `pg_pass_key` is provided.
+- `pg_pass`: The password to connect with. Use the [secret replacement syntax](../../secret-stores/index.md) to load the password from a secret store, e.g. `${secrets:my_pg_pass}`.
 - `pg_sslmode`: Optional parameter, specifies the SSL/TLS behavior for the connection, supported values:
```
Comment (digadeesh) [inline spiceaidocs/docs/components/data-connectors/postgres/index.md]:
```suggestion
- `pg_sslmode`: Optional. Specifies the SSL/TLS behavior for the connection, supported values:
```

File: spiceaidocs/docs/components/data-connectors/flightsql.md [hunk 6]
```diff
@@ -9,45 +9,64 @@ import TabItem from '@theme/TabItem';
...
+    flightsql_password: ${secrets:my_flightsql_pass}
+```
 
-Username and password credentials can be specified to connect to the Flight SQL server:
+## `params`
 
-- `username` (optional): The username to use in the underlying Apache flight Handshake Request to authenticate to the server (see [reference](https://arrow.apache.org/docs/format/Flight.html#authentication)).
-- `password` (optional): The password to use in the underlying Apache flight Handshake Request to authenticate to the server (see [reference](https://arrow.apache.org/docs/format/Flight.html#authentication)).
+- `flightsql_endpoint`: The Apache Flight endpoint used to connect to the Flight SQL server.
+- `flightsql_username` (optional): The username to use in the underlying Apache flight Handshake Request to authenticate to the server (see [reference](https://arrow.apache.org/docs/format/Flight.html#authentication)).
```
Comment (digadeesh) [inline spiceaidocs/docs/components/data-connectors/flightsql.md]:
```suggestion
- `flightsql_username`: Optional. The username to use in the underlying Apache flight Handshake Request to authenticate to the server (see [reference](https://arrow.apache.org/docs/format/Flight.html#authentication)).
```

**Review by digadeesh — DISMISSED** [review]

File: spiceaidocs/docs/components/data-connectors/delta-lake.md [hunk 7]
```diff
@@ -10,127 +10,106 @@ import TabItem from '@theme/TabItem';
...
-  <TabItem value="keyring" label="Keyring">
-    Add new keychain entry (macOS), with secrets in JSON string
+#### AWS S3
 
-    ```bash
-    security add-generic-password -l "Delta Lake Secret" \
-    -a spiced -s spice_secret_delta_lake \
-    -w $(echo -n '{"aws-region": "<aws-region>", "aws-access-key-id": "<aws-access-key-id>", "aws-secret-access-key": "<aws-secret-access-key>"}')
-    ```
+- `databricks_aws_region`: Optional. The AWS region for the S3 object store. E.g. `us-west-2`.
```
Comment (digadeesh) [inline spiceaidocs/docs/components/data-connectors/delta-lake.md]: Should these be databricks_ or delta_lake_?

Comment (phillipleblanc) [reply spiceaidocs/docs/components/data-connectors/delta-lake.md]: Changed to `delta_lake`

**Review by digadeesh — APPROVED** [review]

**Pull request merged**